monai/visualize/class_activation_maps.py

```
@@ -124,19 +124,31 @@ def get_layer(self, layer_id: str | Callable[[nn.Module], nn.Module]) -> nn.Modu
 
-    def class_score(self, logits: torch.Tensor, class_idx: int) -> torch.Tensor:
-        return logits[:, class_idx].squeeze()
+    def class_score(self, logits: torch.Tensor, class_idx: int | torch.Tensor) -> torch.Tensor:
+        if isinstance(class_idx, int):
+            return logits[:, class_idx].squeeze()
+        elif class_idx.numel() == 1:
+            return logits[:, class_idx.item()]
+        elif len(class_idx.view(-1)) == logits.shape[0]:
+            return torch.gather(logits, 1, class_idx.unsqueeze(1)).squeeze(1)
+        else:
+            raise ValueError("expect length of class_idx equal to batch size")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)